### PR TITLE
Added book-keeping class for convergence tests

### PR DESCRIPTION
**Pull request by sebglane**
After #107 .

### CODE REVIEW

File: include/rotatingMHD/convergence_test.h [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+   const std::map<typename VectorTools::NormType, double> &error_map);
+
+  template <int dim, int spacedim>
+  void update_table
+  (const DoFHandler<dim, spacedim>  &dof_handler,
+   const std::map<typename VectorTools::NormType, double> &error_map);
+
+  void update_table
+  (const double time_step,
+   const std::map<typename VectorTools::NormType, double> &error_map);
```
Comment (j507) [inline include/rotatingMHD/convergence_test.h:179]: Documentation

File: include/rotatingMHD/convergence_test.h [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  /*!
+   * @brief Temporal convergence test.
+   * @details Test to study the temporal discretization dependence of
+   * convergence for a given problem.
+   * @note Temporal convergence tests should be performed with a fine
+   * spatial discretization, *i. e.*, a triangulation with small enough cells.
+   */
+  temporal = 0x0002,
+
+  spatio_temporal = spatial|temporal
```
Comment (j507) [inline include/rotatingMHD/convergence_test.h:77]: Documentation

File: source/convergence_test.cc [hunk 3]
```diff
@@ -0,0 +1,684 @@
...
+}
+
+template <int dim>
+void ConvergenceAnalysisData<dim>::update_table
+(const unsigned int level,
+ const double       time_step,
+ const bool         flag_spatial_convergence)
+{
+  /*
+   * Add new entries to the columns describing the spation-temporal
```
Comment (j507) [inline source/convergence_test.cc:56]: Typo?

File: source/convergence_test.cc [hunk 4]
```diff
@@ -0,0 +1,684 @@
...
+                                          VectorTools::H1_norm);
+      convergence_table.add_value("H1", H1_error);
+    }
+
+    /*
+     * For the infinity norm, the quadrature rule is designed such that the
+     * quadrature points coincide with the support points. For a polynomial degree
+     * less than three, the approach pursued here works, because the support
+     * points of the finite element discretization are equidistantly spaced on
+     * the reference element. Otherwise, the quadrature might in appropriate.
```
Comment (j507) [inline source/convergence_test.cc:109]: "might be inappropriate"?

File: source/convergence_test.cc [hunk 5]
```diff
@@ -0,0 +1,684 @@
...
+    case ConvergenceTestType::spatial:
+      column.assign("h_max");
+      evaluate_convergence = true;
+      break;
+    case ConvergenceTestType::temporal:
+      column.assign("step size");
+      dimension = 1;
+      evaluate_convergence = true;
+      break;
+    case ConvergenceTestType::spatio_temporal:
```
Comment (j507) [inline source/convergence_test.cc:464]: I see some `switch` scopes considering this third options and other who does not take it in consideration. Is this by design?

**Review by j507 — CHANGES_REQUESTED** [review]
There are methods and members missing documentation. The option to do both a spatial and a temporal convergence test is introduced but is not considered in some `switch` scopes. The option itself is not defined in the `*.cc` files.

File: source/convergence_test.cc (hunk 5 above)
Comment (sebglane) [reply source/convergence_test.cc:464]: For a spatio-temporal convergence test in `ConvergenceTestData` the only difference is that the convergence rates are not evaluated.

File: source/convergence_test.cc (hunk 4 above)
Comment (sebglane) [reply source/convergence_test.cc]: Fixed in 5164134

File: source/convergence_test.cc (hunk 3 above)
Comment (sebglane) [reply source/convergence_test.cc]: Fixed in 5164134

File: include/rotatingMHD/convergence_test.h (hunk 2 above)
Comment (sebglane) [reply include/rotatingMHD/convergence_test.h:89]: Fixed in 5164134

File: include/rotatingMHD/convergence_test.h (hunk 1 above)
Comment (sebglane) [reply include/rotatingMHD/convergence_test.h]: Fixed in 5164134

File: include/rotatingMHD/convergence_test.h [hunk 6]
```diff
@@ -147,8 +161,77 @@ struct ConvergenceTestParameters
...
+ * @class ConvergenceTestData
+ *
+ * @brief @ref ConvergenceTestData is a book-keeping class for the errors of a convergence test.
+ *
+ */
+class ConvergenceTestData
+{
+
+public:
+
```
Comment (j507) [inline include/rotatingMHD/convergence_test.h:175]: Documentation

File: include/rotatingMHD/convergence_test.h [hunk 7]
```diff
@@ -147,8 +161,77 @@ struct ConvergenceTestParameters
...
+  ConvergenceTable  table;
+
+  bool step_size_specified{false};
+  bool h_max_specified{false};
+  bool L2_error_specified{false};
+  bool H1_error_specified{false};
+  bool Linfty_error_specified{false};
+
+  void format_columns();
```
Comment (j507) [inline include/rotatingMHD/convergence_test.h:230]: Documentation

**Review by j507 — COMMENTED** [review]
I hadn't marked this while reviewing.

File: include/rotatingMHD/convergence_test.h (hunk 7 above)
Comment (sebglane) [reply include/rotatingMHD/convergence_test.h:230]: Fixed in 3536dc4.

File: include/rotatingMHD/convergence_test.h (hunk 6 above)
Comment (sebglane) [reply include/rotatingMHD/convergence_test.h:175]: Constructor doc string is still missing in 3536dc4.

**Pull request merged**